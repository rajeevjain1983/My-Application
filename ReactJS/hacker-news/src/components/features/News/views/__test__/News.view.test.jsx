import React from "react";

import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import NewsView from "../News.view";

configure({ adapter: new Adapter() });

describe("<News", () => {
  it("Render News view", () => {
    const props = {
      news: {
        pageInfo: {
          pageNumber: 1,
          nextPage: 2,
          apiPageNumber: 0,
        },
        newsList: [
          {
            url: "url",
            index: 1,
            title: "test",
            points: 10,
            author: "test",
            created_at: new Date("2020-05-26T06:02:13.382Z"),
            num_comments: 10,
            hide: false,
            upVotes: 0,
            linkDomain: "google.com",
          },
          {
            url: "url",
            index: 1,
            title: "test",
            points: 10,
            author: "test",
            created_at: new Date("2020-05-26T06:02:13.382Z"),
            num_comments: 10,
            hide: false,
            upVotes: 0,
            linkDomain: "google.com",
          },
        ],
        hideComment: () => {},
        upVotes: () => {},
      },
    };
    const component = shallow(<NewsView {...props} />);

    expect(component).toMatchSnapshot();
  });
});
