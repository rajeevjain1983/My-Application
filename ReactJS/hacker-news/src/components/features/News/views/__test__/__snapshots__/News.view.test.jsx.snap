// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<News Render News view 1`] = `
<NewsView
  className="sc-AxhCb cSudCs"
  news={
    Object {
      "hideComment": [Function],
      "newsList": Array [
        Object {
          "author": "test",
          "created_at": 2020-05-26T06:02:13.382Z,
          "hide": false,
          "index": 1,
          "linkDomain": "google.com",
          "num_comments": 10,
          "points": 10,
          "title": "test",
          "upVotes": 0,
          "url": "url",
        },
        Object {
          "author": "test",
          "created_at": 2020-05-26T06:02:13.382Z,
          "hide": false,
          "index": 1,
          "linkDomain": "google.com",
          "num_comments": 10,
          "points": 10,
          "title": "test",
          "upVotes": 0,
          "url": "url",
        },
      ],
      "pageInfo": Object {
        "apiPageNumber": 0,
        "nextPage": 2,
        "pageNumber": 1,
      },
      "upVotes": [Function],
    }
  }
/>
`;
